src/context/TodoContext.tsx

```diff
@@ -0,0 +1,103 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { Todo } from '../types/Todo';
+import { Status } from '../types/Status';
+import { getFilteredTodos } from '../services/getFilteredTodos';
+import { useLocalStorage } from '../hooks/useLocalStorage';
+import { TodoContextType } from '../types/TodoContext';
+import { TodoFilterContextType } from '../types/TodoFilter';
+import { TodoUpdateContextType } from '../types/TodoUpdateContext';
+
+export const TodoContext = React.createContext<TodoContextType>({
+  todos: [],
+  filteredTodos: [],
+});
+
+export const FilterContext = React.createContext<TodoFilterContextType>({
+  status: Status.All,
+  changeStatus: () => { },
+});
+
+export const TodoUpdateContext = React.createContext<TodoUpdateContextType>({
+  addTodo: () => { },
+  completeAll: () => { },
+  deleteCompleted: () => { },
+  deleteTodo: () => { },
+  updateTodo: () => { },
+});
```

Comment:
I think the best way use one context for this task.

Reply:
Can you explain why?

Reply:
Because in this case, you don`t have a different structure. 
All operations work with todo. In this logic, you need to create only one todo context, and all operations with todo will be in one place. Also, context communicates with all components which wrapped. If you can decrease the count of context in a large application, you can increase application productivity.